code/datums/hud.dm

```diff
@@ -35,7 +35,7 @@ GLOBAL_LIST_INIT(huds, list(
 
 /datum/atom_hud
 	var/list/atom/hudatoms = list() //list of all atoms which display this hud
-	var/list/mob/hudusers = list() //list with all mobs who can see the hud
+	var/list/hudusers = list() //list with all mobs who can see the hud

```

Comment:
?

Reply:
hudusers is an assoc list but the value isnt a /mob, only the assoc key is.
`var/list/type/foo` doesnt do anything different in the byond compiler than `var/list/foo`, the typepath after /list exists to tell dreamchecker what type exists in the list's values.
in this case dreamchecker thinks you are doing (/mob)++ below which is a bad use of operator++ since /mob doesnt define `mob/proc/operator++()`
In reality the operation is safely doing (int)++ since the values in the list are ints